atom/browser/api/atom_api_web_contents.cc

```diff
@@ -893,21 +939,13 @@ void WebContents::DidFinishNavigation(
 void WebContents::SecurityStyleChanged(
     content::SecurityStyle security_style,
     const content::SecurityStyleExplanations& explanations) {
-    std::string type = "unknown";
-    switch (security_style) {
-      case content::SECURITY_STYLE_UNAUTHENTICATED:
-      case content::SECURITY_STYLE_AUTHENTICATION_BROKEN:
-        type = "insecure";
-        break;
-      case content::SECURITY_STYLE_WARNING:
-        type = "warning";
-        break;
-      case content::SECURITY_STYLE_AUTHENTICATED:
-        type = "secure";
-        break;
-      default: break;
+    if (explanations.ran_insecure_content) {
+      Emit("security-style-changed", "active-mixed-content");
```

Comment:
@diracdeltas - I think we might be mixing the meanings of these things a little bit. active and passive mixed content are metadata about the current security style. The explanations also have a summary and description that would be useful for the ui so maybe this should emit more fields? Maybe the security style (without passive or active), the summary and description (from explanations.secure_explanations, unauthenticated_explanations or broken_explanations as appropriate to the security style) and then two bools for ran and displayed insecure content? It looks like there can actually be more than one explanation for the current state so maybe we should use a converter to map the explanations to an array of hashes `[{summary: "..", desc: "..."}, ...]`